### PR TITLE
fix: prevents putting meaningless tasks into the queue

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/ADF-270

### CODE REVIEW

File: model/search/ResultsWatcher.php [hunk 1]
```diff
@@ -150,4 +151,14 @@ private function transformDateTime(string $getStartTime): string
 
         return $date->format('m/d/Y H:i:s');
     }
+
+    private function getAdvancedSearchChecker(): AdvancedSearchChecker
+    {
+        return $this->getServiceLocator()->get(AdvancedSearchChecker::class);
+    }
+
+    private function getUserService(): UserService
```
Comment (gabrielfs7) [inline model/search/ResultsWatcher.php:160]: thanks 🙇

File: test/unit/model/search/ResultWatcherTest.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+     */
+    private $advancedSearchChecker;
+    /**
+     * @var MockObject|string
+     */
+    private $queueDispatcher;
+    /**
+     * @var Search|MockObject
+     */
+    private $searchService;
```
Comment (gabrielfs7) [inline test/unit/model/search/ResultWatcherTest.php:55]: PSR + CR agreement + wrong annotation

```suggestion
    /**
     * @var ResultsWatcher
     */
    private $subject;
    
    /**
     * @var AdvancedSearchChecker|MockObject
     */
    private $advancedSearchChecker;
    
    /**
     * @var QueueDispatcherInterface|MockObject
     */
    private $queueDispatcher;
    
    /**
     * @var Search|MockObject
     */
    private $searchService;
```

**Review by gabrielfs7 — COMMENTED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by chinnu-focaloid — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [X Acceptance criteria are respected
- [X] Pull request title and description are meaningful

File: test/unit/model/search/ResultWatcherTest.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+     */
+    private $subject;
+    /**
+     * @var AdvancedSearchChecker|MockObject
+     */
+    private $advancedSearchChecker;
+    /**
+     * @var MockObject|string
+     */
+    private $queueDispatcher;
```
[the remark this answers is not in the document]
Comment (gabrielfs7) [reply test/unit/model/search/ResultWatcherTest.php]: It was already fixed

**Review by gabrielfs7 — APPROVED** [review]

**Pull request merged**